src/Blocks/BlockPiston.cpp

```diff
@@ -228,11 +234,6 @@ bool cBlockPistonHandler::CanPushBlock(
 	NIBBLETYPE currMeta;
 	a_World.GetBlockTypeMeta(a_BlockPos, currBlock, currMeta);
 
-	if (!cChunkDef::IsValidHeight(a_BlockPos.y))
-	{
-		return !a_RequirePushable;
```

Comment:
I didn't really understand `a_RequirePushable`, to me `CanPushBlock` should just return whether the block can be pushed independent on whether we need it to be pushed or not. Maybe @tigerw can shed some light on this.

Reply:
`RequirePushable` should be named` IsPushing`! When retracting, `IsPushing == false` which changes the behaviour of `CanPush`, making it ignore a piston-breakable block instead of killing it. In this case I agree returning false is fine, this will stop extension and leave retraction unaffected.

Reply:
this file could do with some improvements to clarity, the logic is dense...